### ISSUE TITLE
arccos behaves differently on branch cut from numpy and fails for large values

### ISSUE THREAD

**Issue by anntzer** [pydata/numexpr#208]
Briefly:

```
In [15]: a = np.array(1e3+0j); print(np.arccos(a), ne.evaluate("arccos(a)"))              
-7.60090220954j 7.600902209454717j
In [23]: a = np.array(1e8+0j); print(np.arccos(a), ne.evaluate("arccos(a)"))
-19.1138279245j infj
```

(sorry, mistakenly opened a bug here that I wanted to open on numpy also regarding arccos, so I repurposed this for another issue)

**Comment by anntzer** [comment]
Any reason why the issue was closed?  I can still observe it as of numexpr 2.6.2.

**Comment by robbmcleod** [comment]
There's an `arccos` function in the 3.0 development branch,

https://github.com/pydata/numexpr/blob/numexpr-3.0/numexpr3/functions_GENERATED.cpp

on line 10213 and 10241. For complex functions, see,

https://github.com/pydata/numexpr/blob/numexpr-3.0/numexpr3/complex_functions.hpp

on lines 976 and 1009.

If these functions don't meet your standards, feel free to re-open, but there are test functions for them, and they match the NumPy results within the range of tested values within a tolerance of 1e-6. NumExpr is designed for speed so it does make some trade-offs in terms of accuracy.  There's a large back-log of issues and I'm trying to remove those that have been dealt with or never will be dealt with.  This one seems to work as expected.

**Comment by anntzer** [comment]
Thanks, I did not realize that the issue was fixed in 3.0.
